client/app/deploy/DeployController.js

```diff
@@ -0,0 +1,178 @@
+      vm.model.DeploymentMaps.splice(index, 1);
+    }
+
+    vm.doAllTheThings = function () {
+      function createJob(main) {
+        var subs = Array.prototype.slice.call(arguments, 1);
+        return { Name: main, SubTasks: subs.map(function (t) { return t; }) }
+      }
+
+      var portsUsed;
+
+      function createService() {
+        return serviceService.create(vm.model)
+          .then(function (pair) {
+            portsUsed = pair;
+            var serviceCreatedTask = `Created Service (${vm.model.ServiceName})`;
+            var portsCreatedTask = `Did not create any ports (service worker)`
+            if (vm.model.ServiceType.toLowerCase().startsWith('http')){
+              var portsCreatedTask = `Create Service Ports (${portsUsed.Green}, ${portsUsed.Blue})`;
+            }
+            completedJobs.push(createJob(serviceCreatedTask, portsCreatedTask, `Created Service ID(${vm.model.ServiceName})`));
+          });
+      }
+
+      function addServiceToServerRole() {
+        return clientServerRoleService.create(vm.model)
+          .then(function () {
+            completedJobs.push(createJob('Modified Deployment Map', `Added service(${vm.model.ServiceName})`, `Modified roles for service`));
+          });
+      }
+
+      function createUpstream() {
+        if (!vm.model.ServiceType.toLowerCase().startsWith('http')) return Promise.resolve();
+        var promises = [];
+        vm.model.DeploymentMaps.forEach(function (deploymentMap) {
+          promises.push(clientUpstreamService.create(deploymentMap.SelectedEnvironment, vm.model.ServiceName, portsUsed.Blue, portsUsed.Green).then(function () {
+            completedJobs.push(createJob('Created Upstream Settings', `${vm.model.ServiceName} in ${deploymentMap.SelectedEnvironment} on ${portsUsed.Blue}(blue)`, `${vm.model.ServiceName} in ${deploymentMap.SelectedEnvironment} on ${portsUsed.Green}(green)`));
+          }));
+        });
+
+        return Promise.all(promises);
+      }
+
+      function createLoadBalancerSettings() {
+        var promises = [];
+
+        vm.model.DeploymentMaps.forEach(function (deploymentMap) {
+          promises.push(clientLoadBalancerService.create(deploymentMap, deploymentMap.SelectedEnvironment, vm.model.ServiceName).then(function () {
+            completedJobs.push(createJob('Create Load Balancer Settings', `${deploymentMap.loadBalancerUrl}`));
+          }));
+        })
+
+        return Promise.all(promises);
+      }
+
+      var completedJobs = [];
+
+      return Promise.all([createService()])
+        .then(function () {
+          return Promise.all([
+            addServiceToServerRole(),
+            createUpstream(),
+            createLoadBalancerSettings()
+          ]);
+        })
+        .then(function () { vm.result = completedJobs; $scope.$apply(); console.log(vm.result) });
+    };
+
+    vm.finishedWizard = function () {
+    };
+
+    vm.cancelledWizard = function () {
```

Comment:
Can vm.finishedWizard and vm.cancelledWizard be deleted?

Reply:
Yes sure